### PR TITLE
fix(web): red error bubble for assistant error messages in MinimalThreadFeed

### PR DESCRIPTION
**Pull request by lsm**
## What

In the Space task thread, an assistant message carrying an SDK `error` field (e.g. `error: 'invalid_request'`, the "API Error: 400 …" text the user sees) rendered as a **normal gray reply bubble** — indistinguishable from a successful reply. The chat transcript already renders this correctly (`SDKAssistantMessage` detects `hasError` and swaps to a red bubble), but `MinimalThreadFeed` never inspected the `error` field.

## Change

- **`extractLastAssistantText`**: detect the assistant `error` flag (`'error' in message && error !== undefined`) on the surfaced assistant row, and track it across all assistant rows so the flag still propagates when text comes from a result/tail fallback. Returns a new `hasError` boolean.
- **`CompletedFeedTurn`**: carries the `hasError` flag.
- **`CompletedBody`**: when `hasError`, swaps the bubble to red (`bg-red-900/20 border-red-800`, matching the dark-mode palette of the always-dark thread feed) with an "API Error" header (warning icon + label), mirroring `SDKAssistantMessage`'s `hasError` branch. Body text uses `text-red-100`.

## Acceptance

A 400/invalid_request assistant message now shows a red error bubble in the space thread (not a gray reply), matching the chat transcript's styling.

## Test

- New: assistant error message renders a red bubble (`data-has-error`, `border-red-800`, `bg-red-900`, "API Error" header) with the error text as the body.
- New: normal assistant reply keeps the gray bubble (`bg-dark-800`, no error styling).
- All 65 tests in `MinimalThreadFeed.test.tsx` pass.

### CODE REVIEW

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude (Anthropic)

> **Model:** Claude | **Client:** NeoKai | **Provider:** Anthropic

Recommendation: APPROVE

Reviewed the diff, ran the focused test file (65/65 passing), and ran the full repo `check` (lint/typecheck/knip/session-guards/space-task/db-schema/test-quality all clean; the only lint warnings are pre-existing in `packages/daemon/src/lib/space/workflows/workflow-migration.ts`).

Findings: none.

- `extractLastAssistantText` correctly detects the assistant `error` field using the same `"error" in message && message.error !== undefined` predicate as `SDKAssistantMessage`, and propagates `hasError` through result/tail fallbacks.
- `CompletedBody` uses the dark-mode red palette (`bg-red-900/20`, `border-red-800`, `text-red-400`/`text-red-100`) appropriate for the always-dark thread feed, with the same warning icon + "API Error" header as the chat transcript.
- Tests cover both the error bubble and the regression guard for normal gray bubbles.
- No scope creep; interface changes are localized to this file.

File: packages/web/src/components/space/thread/minimal/MinimalThreadFeed.tsx [hunk 1]
```diff
@@ -589,17 +611,23 @@ function extractLastAssistantText(rows: ParsedThreadRow[]): {
       )
       .map((block) => block.text.trim())
       .filter((s) => s.length > 0);
-    if (texts.length > 0) return { text: texts.join('\n\n'), fallback: false, sourceRow: row };
+    if (texts.length > 0)
+      return { text: texts.join('\n\n'), fallback: false, sourceRow: row, hasError };
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/web/src/components/space/thread/minimal/MinimalThreadFeed.tsx:631]: Thanks for the careful read. I considered this, but the row-local `hasError` on the early return is intentional — accumulating would cause a worse regression.

The scenario described (earlier `error` row + later normal-text row in the same completed turn) can only arise when the turn **recovered**: the SDK hit a transient error (e.g. 429/overloaded) and retried successfully, producing a normal assistant reply afterward. `extractLastAssistantText` surfaces the **last** reply by design (reverse scan), so `lastMessage` is the agent's *successful* final reply. In that state the gray success bubble is correct — the turn's terminal state is success.

If the early return used the accumulated flag (`hasError || assistantHasError`), a recovered turn would render a **red "API Error" bubble whose body text is a successful reply** — that is strictly more misleading than the current behavior.

For the gap this change targets (400 invalid_request), the error is **terminal** — it is the last assistant message in the turn, so the reverse scan returns it directly with `hasError = true` (covered by `renders an assistant error message as a red error bubble`).

The accumulated `assistantHasError` **is** wired into the two fallback paths (result-success text and tail fallback), which covers the edge case where the error row has no text of its own.

I added a test (`renders a recovered turn (error then success) as a gray bubble, not red`, ec48ec4) to document this as intended behavior and guard against a future regression in the wrong direction.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude (Anthropic)

> **Model:** Claude | **Client:** NeoKai | **Provider:** Anthropic

Recommendation: APPROVE

Re-reviewed after the update. The implementation is unchanged and correct; the only addition is a regression test locking in the row-local `hasError` semantics.

- Verified 66/66 tests pass in `MinimalThreadFeed.test.tsx`.
- Verified full repo `check` is clean (lint/typecheck/knip/session-guards/space-task/db-schema/test-quality); remaining lint warnings are pre-existing in `workflow-migration.ts`.
- Verified the prior review thread is resolved.
- The recovered-turn test (`error` assistant row followed by a successful `assistantText` row) confirms the bubble stays gray, which is the right product call: the surfaced `lastMessage` is the successful reply, so a red "API Error" header would be misleading.

No findings. Approve.
